wallet/src/actors/app/params.rs

```diff
@@ -16,6 +20,25 @@ pub struct Params {
 }
 
 pub struct NodeClient {
-    pub url: String,
     pub actor: Addr<JsonRpcClient>,
+    pub url: Arc<Mutex<String>>,
+}
+
+impl NodeClient {
+    /// Get the URL that the current client is connecting to.
+    pub fn current_url(&self) -> String {
+        self.url.lock().unwrap().to_string()
```

Comment:
This function is only used in `notify_client` to get the address of the current node. Instead of using a mutex we could ask the `actor` for the current address, but that would make this function `async`, and that would be a problem because the `notify_client` function is not `async`. However looking at the body of `notify_client`, there is a `futures::executor::block_on` which indicates that that function should actually be `async`.

So may be worth it to try to refactor this method into a `self.actor.send(GetCurrentUrl).await` or similar. This way we would avoid the mutex.

Reply:
Good point. I used the mutex in the first place to avoid sending messages, handling futures, etc.

So it all boils down to whether we prefer to use mutexes to save us from actor overhead, or the other way around hehehe.

Reply:
In this case the natural pattern is to just send messages I believe, because it is the `JsonRpcClient` actor who owns the `url`. If we are using a mutex then the `NodeClient` is technically allowed to change the url, but it shouldn't be, it should only be able to read it. And not sure about overhead but the actor messages are implemented using similar mutexes probably.